### ISSUE TITLE
No more than 3 controllers work if steam input is enabled

### ISSUE THREAD

**Issue by marco-calautti** [ValveSoftware/steam-for-linux#9571]
#### Your system information

* Steam client version (build number or date): 1685572056
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
If Steam input is enabled, when I try to play games supporting 4 players or more, the games only recognize 3 of the connected controllers. If I disable steam input, then (given the game has built-in support for the controllers I have), the game recognizes all controllers.

#### Steps for reproducing this issue:

1. Enable steam input and enable support for all gamepad models you have
2. Start a game supporting 4 players or more
3. Start a 4+ players game

This issue essentially makes impossible playing 4+ players games that have no built-in support for gamepads (e.g., DS4 or Pro Controller). Which means people pay for a game that they will not be able to play as advertised.

Steam input is broken in so many ways that it is almost embarassing. If Valve has not or does not want to invest resources to fix the Steam input system (which seems to be so, according to the plenty of open and unresolved issues on this repository), then I would suggest open sourcing at least the Steam input component of Steam, to let the community jump in, in the same way Valve is doing with Proton.

**Comment by TTimo** [comment]
Hello @marco-calautti, can you provide a game that you have tested this with, and what assortment of controllers did you use?

**Comment by marco-calautti** [comment]
> It's worth noting that Brawlhalla has a hardcoded limit of 4 controllers. If the host has more than 4 controllers plugged in, all extras will be ignored by the game (this can be verified on Windows as well). This matters on Steam Deck where you will need to rearrange controller order so the builtin Deck controller doesn't get picked up first (we tested Deck to be working as well).

This might probably be the issue, as I have permanently connected to usb dongles for the Guitar Hero Guitars, which probably get picked first. I will try switching the order, and will report back.

**Comment by marco-calautti** [comment]
It seems that changing the order of the controllers made the game recognize them. 
However, I find this behaviour very counter-intuitive and strictly related to how the OS/steam connects to the controllers.
I would expect the game to pick up not  the first 4 controllers in a list given by the OS/Steam, but the first 4 controllers that provide for the first time some input to the game, similarly to what happens in a console. Steam input could do something in this regard to make this process smoother, and similar to what an actual gaming console does.

**Comment by Junohea** [comment]
Having this issue as well in any 4player game tested, including
- Helldivers
- Ember Knights
- Castle Crashes
- Screen Cheat
- Hammerwatch
and i suspect the same across most other games, it doesnt seem to matter how many ways i reorder the controllers either.

Tried with a mix of wired and wireless controllers, across a mix of Xbox One and Xbox Series X controllers

This was attempted with the official wireless dongle and using https://github.com/medusalix/xone

if there's any additional information i can provide I'm happy to run some commands or help how i can

**Comment by marco-calautti** [comment]
As I was suggested, change the controller order in the game's controller settings. The game will pick up the first 4 controllers of the list.

**Comment by Ptkstefano** [comment]
> As I was suggested, change the controller order in the game's controller settings. The game will pick up the first 4 controllers of the list.

I wasn't able to get the game to not pickup the Steam Deck controller regardless of order in the settings and I tried changing the order multiple times and relaunching the game. I'll try again anyway.

**Comment by pwheaton** [comment]
> As I was suggested, change the controller order in the game's controller settings. The game will pick up the first 4 controllers of the list.

This didn't work for me in any of the four player games I tried:
- towerfall
- ember knights
- assault android cactus 
- circuit superstars
- new star GP

I find rebooting the deck can resolve the issue temporarily. Usually for one game. Switching to a new game brings the issue back. 

I have also tried a variety of wired vs wireless, all controllers work fine at the OS.

If I'm playing the deck while on the official dock there is no situation where I'm using the deck as a controller.
